components/script/dom/htmlvideoelement.rs

```diff
@@ -416,3 +473,27 @@ impl PosterFrameFetchContext {
         }
     }
 }
+
+pub trait LayoutHTMLVideoElementHelpers {
+    fn data(self) -> HTMLMediaData;
+}
+
+impl LayoutHTMLVideoElementHelpers for LayoutDom<'_, HTMLVideoElement> {
+    fn data(self) -> HTMLMediaData {
+        let video = self.unsafe_get();
+
+        if let Some(frame) = video.htmlmediaelement.get_current_frame_data() {
+            return HTMLMediaData {
+                current_frame: Some(frame.image_key),
+                width: frame.width,
+                height: frame.height,
+            };
+        }
+
+        HTMLMediaData {
+            current_frame: None,
+            width: video.video_width.get() as i32,
+            height: video.video_height.get() as i32,
+        }
+    }
```

Comment:
This is mixing the natural sizes of the video with the sizes from the attributes (which should just set CSS presentational hints). Then I think this will break `object-fit`.

Reply:
Ok, that makes a lot of sense, now I see it. I'll make some changes